### PR TITLE
fixes #4108

### LINKED ISSUE #4108: Allow user to remove custom avatar
### Description

*Type*: Bug

*Summary*: As a user i want to be able to remove my custom avatar (photo) and restore it to default.

#### Expected behavior
User is able to revert avatar to default one that was created with the account. @denis-sharypin is it possible to add some design?

#### Actual behavior
Once avatar is changed to picture or photo user is not able to restore the default value 

### Reproduction
- Open Status
- Tap profile
- Tap edit 
- Change avatar and save

### Additional Information
* Status version: nightly
* Operating System: Android, iOS

### PR DESCRIPTION
**Pull request by 751895405**
fixes #4108 Allow user to remove custom avatar

### CODE REVIEW

File: src/status_im/ui/screens/profile/user/views.cljs [hunk 1]
```diff
@@ -206,14 +218,24 @@
...
         [react/view profile.components.styles/profile-form
          [profile.components/profile-header
           {:contact              current-account
            :edited-contact       changed-account
            :editing?             editing?
            :allow-icon-change?   true
            :options              profile-icon-options
-           :on-change-text-event :my-profile/update-name}]]
+           :on-change-text-event :my-profile/update-name}]])
+        (when (not= (identicon/identicon public-key) photo-path)
```
Comment (flexsurfer) [inline src/status_im/ui/screens/profile/user/views.cljs]: why is it duplicated?

File: src/status_im/ui/screens/profile/user/views.cljs [hunk 2]
```diff
@@ -206,14 +218,24 @@
          [my-profile-toolbar])
        [react/scroll-view {:ref                          #(reset! scroll %)
                            :keyboard-should-persist-taps :handled}
+        (when (= (identicon/identicon public-key) photo-path) 
         [react/view profile.components.styles/profile-form
```
Comment (flexsurfer) [inline src/status_im/ui/screens/profile/user/views.cljs]: please fix the indentation
```
(when (= (identicon/identicon public-key) photo-path) 
   [react/view profile.components.styles/profile-form	        
```

File: src/status_im/ui/screens/profile/user/views.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/status_im/ui/screens/profile/user/views.cljs]:
```
:options          (if (= (identicon/identicon public-key) photo-path)     
                           profile-icon-options
                           profile-icon-options-ext
```

File: src/status_im/ui/screens/profile/user/views.cljs [hunk 3]
```diff
@@ -45,6 +46,17 @@
                                                    :icon                :icons/ok
                                                    :icon-opts           {:color colors/blue}
                                                    :accessibility-label :done-button}]])}))
+(def profile-icon-options-ext
```
Comment (flexsurfer) [inline src/status_im/ui/screens/profile/user/views.cljs]: why is it a new function?

**Review by flexsurfer — CHANGES_REQUESTED** [review]
thanks @751895405 for your contribution, i left some comments

**Comment by 751895405** [comment]
i update it

File: src/status_im/ui/screens/profile/events.cljs [hunk 4]
```diff
@@ -48,6 +49,14 @@
             (assoc-in [:my-profile/profile :valid-name?] (valid-name? name))
             (assoc-in [:my-profile/profile :name] name))}))
 
+(handlers/register-handler-fx
+ :my-profile/remove-current-photo
+ (fn [{:keys [db]}]
+  {:db       (-> db
```
Comment (flexsurfer) [inline src/status_im/ui/screens/profile/events.cljs]: could you please fix the formatting for this map

File: src/status_im/ui/screens/profile/user/views.cljs [hunk 5]
```diff
@@ -57,7 +57,10 @@
                                                                        (utils/set-timeout
                                                                         #(utils/show-popup (i18n/label :t/error)
                                                                                            (i18n/label :t/camera-access-error))
-                                                                        50))}]))}])
+                                                                        50))}]))}
+  (when (not= (identicon/identicon public-key) photo-path)
+    {:label  (i18n/label :t/image-remove-current)
+    :action #(re-frame/dispatch [:my-profile/remove-current-photo])})])
```
Comment (flexsurfer) [inline src/status_im/ui/screens/profile/user/views.cljs]: and here as well

**Review by flexsurfer — APPROVED** [review]

**Review by dmitryn — APPROVED** [review]

**Comment by dmitryn** [comment]
You could check formatting with `lein cljfmt check` and fix it with `lein cljfmt fix`. You need to do it anyway since Jenkins build is failing https://jenkins.status.im/blue/organizations/jenkins/status-react%2Fpull%20requests/detail/PR-5177/1/pipeline

<blockquote><div><strong><a href="https://jenkins.status.im/blue/organizations/jenkins/status-react%2Fpull%20requests/detail/PR-5177/1/pipeline">Jenkins</a></strong></div></blockquote>

File: src/status_im/ui/screens/profile/events.cljs (hunk 4 above)
Comment (751895405) [reply src/status_im/ui/screens/profile/events.cljs]: I will fix the formatting very soon thanks

File: src/status_im/ui/screens/profile/user/views.cljs (hunk 5 above)
Comment (751895405) [reply src/status_im/ui/screens/profile/user/views.cljs]: ok

**Review by flexsurfer — APPROVED** [review]

**Comment by lukaszfryc** [comment]
Hey @751895405! Thanks for your contribution. I'm testing it on iPhone 8 (iOS 11.4) and every time when I click "Edit" on the avatar the app crashes. Here is the stacktrace:
```
Application Specific Information:
*** Terminating app due to uncaught exception 'RCTFatalException: Exception '-[NSNull length]: unrecognized selector sent to instance 0x1b3f5f878' was thrown while invoking showActionSheetWithOptions on target ActionSheetManager with params (
        {
        cancelButtonIndex = 3;
        options =         (
            "Select from gallery",
            Capture,
            "",
            Cancel
        );
        title = "Edit picture";
    },
    3817
)
callstack: (
	0   CoreFoundation                      0x000000018192ada4  + 252
	1   libobjc.A.dylib                     0x0000000180ae45ec objc_exception_throw + 56
	2   CoreFoundation                      0x0000000181938098  + 0
	3   CoreFoundation                      0x00000001819305c8  + 1380
	4   CoreFoundation                      0x000000018181641c _CF_forwarding_prep_0 + 92
	5   UIKit                               0x000000018b76c9ac  + 180
	6   UIKit                               0x000000018b76c8b0  + 148
	7   UIKit                               0x000000018b76c7e8  + 108
	8   StatusIm                            0x0000000105baed9c _ZN7plcrash10TestFairy_5async24dwarf_cfa_state_iteratorIyxE4nextEPjPNS1_28plcrash_dwarf_cfa_reg_rule_tEPy + 10006448
	9   CoreFoundation                      0x0000000181932580  + 144
	10  CoreFoundation                      0x0000000181811748  + 284
	11  CoreFoundation                      0x000000018181656c  + 60
	12  StatusIm                            0x000000010506655c StatusIm + 566620
	13  StatusIm                            0x00000001050abd70 RCTFBQuickPerformanceLoggerConfigureHooks + 108508
	14  StatusIm                            0x00000001050abad0 RCTFBQuickPerformanceLoggerConfigureHooks + 107836
	15  libdispatch.dylib                   0x000000018121caa0  + 24
	16  libdispatch.dylib                   0x000000018121ca60  + 16
	17  libdispatch.dylib                   0x000000018125dd80  + 964
	18  CoreFoundation                      0x00000001818d3070  + 12
	19  CoreFoundation                      0x00000001818d0bc8  + 2272
	20  CoreFoundation                      0x00000001817f0da8 CFRunLoopRunSpecific + 552
	21  GraphicsServices                    0x00000001837d6020 GSEventRunModal + 100
	22  UIKit                               0x000000018b810758 UIApplicationMain + 236
	23  StatusIm                            0x0000000104fe33c4 StatusIm + 29636
	24  libdyld.dylib                       0x0000000181281fc0  + 4
)', reason: 'Exception '-[NSNull length]: unrecognized selector sent to instance 0x1b3f5...'
```

Could you fix it?

**Comment by lukaszfryc** [comment]
@751895405 also, please rebase this PR.

**Comment by 751895405** [comment]
I test on android phone it works fine ,I think i should test on my iphone also to check it .

**Comment by lukaszfryc** [comment]
@751895405 I checked latest builds:

> apk uploaded to https://i.diawi.com/oajCaX
ipa uploaded to https://i.diawi.com/xa1YkP

And the app still crashes as described in https://github.com/status-im/status-react/pull/5177#issuecomment-406563737

<blockquote><div><strong><a href="https://i.diawi.com/oajCaX">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://i.diawi.com/xa1YkP">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by lukaszfryc** [comment]
@751895405 I'm testing on iPhone 8, 11.4.1 and the iOS app crashes right after I open it:

```
Incident Identifier: D441CF42-514C-4008-A8A2-2837A73973E1
CrashReporter Key:   d165c86239a26d6e807f116001103cdca0e5c7a7
Hardware Model:      iPhone10,4
Process:             StatusIm [5212]
Path:                /private/var/containers/Bundle/Application/279FFF33-7DE4-4C4A-B06A-AC770D9BB961/StatusIm.app/StatusIm
Identifier:          im.status.ethereum
Version:             6273 (0.9.24)
Code Type:           ARM-64 (Native)
Role:                Foreground
Parent Process:      launchd [1]
Coalition:           im.status.ethereum [726]


Date/Time:           2018-07-30 14:55:13.3420 +0200
Launch Time:         2018-07-30 14:55:13.3177 +0200
OS Version:          iPhone OS 11.4.1 (15G77)
Baseband Version:    1.04.80
Report Version:      104

Exception Type:  EXC_CRASH (SIGABRT)
Exception Codes: 0x0000000000000000, 0x0000000000000000
Exception Note:  EXC_CORPSE_NOTIFY
Termination Description: DYLD, Library not loaded: @rpath/RSKImageCropper.framework/RSKImageCropper | Referenced from: /var/containers/Bundle/Application/279FFF33-7DE4-4C4A-B06A-AC770D9BB961/StatusIm.app/StatusIm | Reason: no suitable image found.  Did find: | /private/var/containers/Bundle/Application/279FFF33-7DE4-4C4A-B06A-AC770D9BB961/StatusIm.app/Frameworks/RSKImageCropper.framework/RSKImageCropper: code signing blocked mmap() of '/private/var/containers/Bundle/Application/279FFF33-7DE4-4C4A-B06A-AC770D9BB961/StatusIm.app/Frameworks/RSKImageCropper.framework/RSKImageCropper'
Triggered by Thread:  0

Filtered syslog:
None found

Thread 0 Crashed:
0   dyld                          	0x00000001029362f8 __abort_with_payload + 8
1   dyld                          	0x0000000102935ad0 abort_with_payload_wrapper_internal + 100
2   dyld                          	0x0000000102935afc fcntl + 0
3   dyld                          	0x000000010290c314 dyld::fastBindLazySymbol+ 17172 (ImageLoader**, unsigned long) + 0
4   dyld                          	0x000000010290f4cc dyld::_main+ 29900 (macho_header const*, unsigned long, int, char const**, char const**, char const**, unsigned long*) + 7152
5   dyld                          	0x000000010290921c _dyld_start + 68

Thread 0 crashed with ARM Thread State (64-bit):
    x0: 0x0000000000000006   x1: 0x0000000000000001   x2: 0x000000016f77e930   x3: 0x00000000000000a3
    x4: 0x000000016f77e530   x5: 0x0000000000000000   x6: 0x0000000000000000   x7: 0x0000000000000ea0
    x8: 0x0000000000000020   x9: 0x0000000000000009  x10: 0x6567616d494b5352  x11: 0x2e726570706f7243
   x12: 0x6172662e72657070  x13: 0x522f6b726f77656d  x14: 0x436567616d494b53  x15: 0x0027726570706f72
   x16: 0x0000000000000209  x17: 0x0000000000000001  x18: 0x0000000000000000  x19: 0x0000000000000000
   x20: 0x000000016f77e530  x21: 0x00000000000000a3  x22: 0x000000016f77e930  x23: 0x0000000000000001
   x24: 0x0000000000000006  x25: 0x0000000100680000  x26: 0x0000000102946000  x27: 0x0000000102946458
   x28: 0x0000000100680020   fp: 0x000000016f77e500   lr: 0x0000000102935ad0
    sp: 0x000000016f77e4c0   pc: 0x00000001029362f8 cpsr: 0x00000000

Binary Images:
0x100680000 - 0x1017e7fff StatusIm arm64  <3bad5d2c8f32382ea86da10319f6f2f8> /var/containers/Bundle/Application/279FFF33-7DE4-4C4A-B06A-AC770D9BB961/StatusIm.app/StatusIm
0x102908000 - 0x102943fff dyld arm64  <b15e536a710732dabfafece44c5685e4> /usr/lib/dyld

EOF
```

**Comment by lukaszfryc** [comment]
@751895405 it works great now 👍 Could you update the [changelog](https://github.com/status-im/status-react/blob/develop/CHANGELOG.md)? Please add "Allow user to remove custom avatar" or something like that under Added section.

@751895405 is the avatar removed from the app storage when user removes it in the app? If not, can you implement this?

**Comment by 751895405** [comment]
avatar has bean encoded by base64 String once you update or remove it by default which will be update on db. the avatar image did not store anywhere else.

**Comment by flexsurfer** [comment]
@751895405 could you please squash commits, thanks

**Comment by flexsurfer** [comment]
sorry @751895405 could you please resolve conflicts, thanks

**Comment by 751895405** [comment]
@flexsurfer I am working on it

**Comment by flexsurfer** [comment]
@751895405 could you please squash commits, thanks

**Comment by flexsurfer** [comment]
@751895405 we're using script to merge PRs, and if there is more than 1 commit it's impossible to merge it, could you please squash commits, thanks

**Pull request closed without merging**